src/go/k8s/pkg/console/deployment.go

```diff
@@ -0,0 +1,342 @@
+package console
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	labels "github.com/redpanda-data/redpanda/src/go/k8s/pkg/labels"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/resources"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Deployment is a Console resource
+type Deployment struct {
+	client.Client
+	scheme     *runtime.Scheme
+	consoleobj *redpandav1alpha1.Console
+	clusterobj *redpandav1alpha1.Cluster
+	store      *Store
+	log        logr.Logger
+}
+
+// NewDeployment instantiates a new Deployment
+func NewDeployment(cl client.Client, scheme *runtime.Scheme, consoleobj *redpandav1alpha1.Console, clusterobj *redpandav1alpha1.Cluster, store *Store, log logr.Logger) *Deployment {
+	return &Deployment{
+		Client:     cl,
+		scheme:     scheme,
+		consoleobj: consoleobj,
+		clusterobj: clusterobj,
+		store:      store,
+		log:        log,
+	}
+}
+
+// Ensure implements Resource interface
+func (d *Deployment) Ensure(ctx context.Context) error {
+	sa, err := d.ensureServiceAccount(ctx)
+	if err != nil {
+		return err
+	}
+
+	ss, err := d.ensureSyncedSecrets(ctx)
```

Comment:
is this needed only if we're running in multiple namespaces? or not?

Reply:
There are two Secrets needed by SchemaRegistry:
1. mTLS client certs
2. CA cert

mTLS cert is on secret `<cluster-name>-schema-registry-client`, CA cert should depend if IssuerRef, NodeSecretRef is present. Provisioning a v1 Cluster sets NodeSecretRef although we are issuing via lets-encrypt.

Although we don't need to copy the Secret if creating in the same namespace. This function copies the two Secrets into one. It's a big refactor to do this right now depending if AllowedConsoleAnyNamespace flag is set. I think I would defer doing that refactor once we finalized if we need the flag or not. WDYT?